storm-server/src/main/java/org/apache/storm/scheduler/ISchedulingState.java

```diff
@@ -271,6 +273,13 @@ boolean wouldFit(
      */
     double getClusterTotalMemoryResource();
 
+    /**
+     * Get the total amount of generic resources (excluding CPU and memory) in cluster.
+     */
+    default Map<String, Double> getClusterTotalGenericResource() {
```

Comment:
Method name should probably be `getClusterTotalGenericResources`  with s

Reply:
Cluster should have a private variable that avoids recalculating this often. 
Similarly, it should have Set<String> genericResourceTypes.
And method `getClusterGenericResourceTypes`

Reply:
Addressed